toolchain/lower/testdata/function/generic/call_different_specific.carbon

```diff
@@ -0,0 +1,149 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+// TIP: To test this file alone, run:
+// TIP:   bazel test //toolchain/testing:file_test --test_arg=--file_tests=toolchain/lower/testdata/function/generic/call_different_specific.carbon
+// TIP: To dump output, run:
+// TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/lower/testdata/function/generic/call_different_specific.carbon
+
+// Builds on call_basic.carbon and call_impl_function.carbon.
+// Test that when equivalence is found for the specific fingerprint, we don't
+// conclude "equivalent" for all the call graph.
+
+fn A[U:! type](x: U) -> U {
+  return x;
+}
+
+fn D[U:! type](x: U) -> U {
+  return x;
+}
+
+fn B[U:! type](x: U*) {
+  D(*x);
+}
+
+fn F[T:! type](x: T*) {
+  A(i32);
+  B(x);
+}
+
+fn M() {
```

Comment:
Could we add a comment in the test somewhere about what we do/don't expect to see in the resulting output?